tests/common/plugins/ptfadapter/ptfadapter.py

```diff
@@ -88,3 +91,27 @@ def reinit(self, ptf_config=None):
         """
         self.kill()
         self._init_ptf_dataplane(self.ptf_ip, self.ptf_nn_port, self.device_num, self.ptf_port_set, ptf_config)
+
+    def update_payload(self, pkt):
+        if isinstance(pkt, scapy.Ether):
+            for proto in (scapy.UDP, scapy.TCP):
+                if proto in pkt:
+                    pkt[proto].load = self._update_payload(pkt[proto].load)
+        elif isinstance(pkt, mask.Mask):
+            for proto in (scapy.UDP, scapy.TCP):
+                if proto in pkt.exp_pkt:
+                    pkt.exp_pkt[proto].load = self._update_payload(pkt.exp_pkt[proto].load)
+        return pkt
+
+    def _update_payload(self, payload):
```

Comment:
Can you add a docstring on what is intended in this method?

Reply:
Sure!